### DISCUSSION TITLE
blog/distributed-caching-in-aspnet-core-with-redis/

### DISCUSSION THREAD

**Discussion by giscus** [codewithmukesh/discussions#17] [General]
# blog/distributed-caching-in-aspnet-core-with-redis/

Let's build an ASP.NET Core application that caches frequently accessed data to a Redis instance, which runs locally on a Docker Container. Redis is the most popular distributed cache provider that offers robust features and scalability options, making it suitable for various applications and deployment scenarios.

https://codewithmukesh.com/blog/distributed-caching-in-aspnet-core-with-redis/?utm_source=newsletter&utm_medium=newsletter&utm_campaign=distributed-caching-in-asp-net-core-with-redis

**Comment by totti240282** [comment]
IHybridCache in .NET 9

**Comment by DotNetTitan** [comment]
Docker for .NET Developers

**Comment by tilamit** [comment]
Hello Mukesh, I was going through this and curious to know if we can configure multiple instances of the Redis via Docker. Suppose, in your post, you configured the Redis as follows:

builder.Services.AddStackExchangeRedisCache(options =>
{
    options.Configuration = "localhost";
    options.ConfigurationOptions = new StackExchange.Redis.ConfigurationOptions()
    {
        AbortOnConnectFail = true,
        EndPoints = { options.Configuration }
    };
});

For two Redis instances via Docker, can we write something for localhost as follows:

builder.Services.AddStackExchangeRedisCache("redis1", options =>
{
    options.Configuration = "localhost";
    options.ConfigurationOptions = new StackExchange.Redis.ConfigurationOptions()
    {
        AbortOnConnectFail = true,
        EndPoints = { options.Configuration }
    };
});

builder.Services.AddStackExchangeRedisCache("redis2", options =>
{
    options.Configuration = "localhost";
    options.ConfigurationOptions = new StackExchange.Redis.ConfigurationOptions()
    {
        AbortOnConnectFail = true,
        EndPoints = { options.Configuration }
    };
});

One more thing, in this case is it possible to put the endpoints directly in this section -  EndPoints = { options.Configuration }? Any sample would be appreciated. Thanks.
